lib/Sema/CSSimplify.cpp

```diff
@@ -11518,6 +11663,36 @@ ConstraintSystem::SolutionKind ConstraintSystem::simplifyFixConstraint(
   }
 
   case FixKind::AllowArgumentTypeMismatch: {
+    // If this is after the code completion location, give a fixed penalty
+    // with the same impact as that for the fix to ignore arguments after the
+    // completion position when matching arguments.
+    if (isForCodeCompletion()) {
+      auto getArgIndex = [&](ConstraintLocator *loc) -> Optional<unsigned> {
```

Comment:
Could you please make it work on `ConstraintLocatorBuilder` instead of allocating a new locator to extract index?